### ISSUE TITLE
CollectionView with AdaptiveTrigger not rendered properly

### ISSUE THREAD

**Issue by unombun** [dotnet/maui#21719]
### Description

On Windows using an AdaptiveTrigger that modifies a Grid that contains a CollectionView, in some cases the collectionView is not rendered as it should

### Steps to Reproduce

1. Create a UI containing a Grid with 2 VisualStates that uses AdaptiveTrigger
2. One state should define that the Grid has 2 columns, the other state says that the Grid has 2 rows
3. In one column/row there should be a collection view
4. Once the windows is resized and the state is changed, navigate to other page
5. From the new page go back 
6. In the previous page the wrong state is used
7. Once a small resize of the window is done, the UI is refreshed as it should

### Link to public reproduction project repository

https://github.com/flesarradu/AdaptiveCollectionView

### Version with bug

8.0.14 SR3.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Did you find any workaround?

No

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 2(8.0.7&8.0.14&8.0.20). Can repro on windows platform with sample project.
Before:
![image](https://github.com/dotnet/maui/assets/77434865/c8c5c3d1-410f-4808-807a-84d5bcc46b10)
After:
![image](https://github.com/dotnet/maui/assets/77434865/46cd4937-13e1-4195-9f97-bfe051bbef69)

**Comment by BenBtg** [comment]
Can confirm as reproducible. Also noticed that even after small resize which corrects the layout, clicking anywhere on the page breaks the layout again. Subsequently resizing fixes the layout permanantly. i.e. further clicks do not break it again.

**Comment by mattleibow** [comment]
This appears to have nothing to do with CollectionView and I can repro in a plain page:


```xaml

<ContentView>
    <ContentView.ControlTemplate>
        <ControlTemplate>

            <Grid x:Name="gridThing" Margin="0,16,0,0">

                <VisualStateManager.VisualStateGroups>
                    <VisualStateGroupList>
                        <VisualStateGroup>
                            <VisualState x:Name="Small">
                                <VisualState.StateTriggers>
                                    <AdaptiveTrigger MinWindowWidth="0" />
                                </VisualState.StateTriggers>
                                <VisualState.Setters>
                                    <Setter Property="Grid.RowDefinitions" TargetName="gridThing" Value="*,*" />
                                    <Setter Property="Grid.Row" TargetName="greenThing" Value="0" />
                                    <Setter Property="Grid.Row" TargetName="redThing" Value="1" />
                                </VisualState.Setters>
                            </VisualState>
                            <VisualState x:Name="Normal">
                                <VisualState.StateTriggers>
                                    <AdaptiveTrigger MinWindowWidth="900" />
                                </VisualState.StateTriggers>
                                <VisualState.Setters>
                                    <Setter Property="Grid.ColumnDefinitions" TargetName="gridThing" Value="*,*" />
                                    <Setter Property="Grid.Column" TargetName="greenThing" Value="0" />
                                    <Setter Property="Grid.Column" TargetName="redThing" Value="1" />
                                </VisualState.Setters>
                            </VisualState>
                        </VisualStateGroup>
                    </VisualStateGroupList>
                </VisualStateManager.VisualStateGroups>

                <Button x:Name="greenThing" Background="Green" />

                <Button x:Name="redThing" Background="Red" />

            </Grid>

        </ControlTemplate>
    </ContentView.ControlTemplate>
</ContentView>
```

No need for navigation either. Just resize the window, and then switch to a new app, anything. I just Alt+Tab to VS and I can see in the live previewer that it just switches back.

**Comment by mattleibow** [comment]
I found the cause of the issue and it is because of strings, states and sames.

If you look at the visual states, there are 2: Small and Normal.

The issue here is that the "Normal" state is too common and is a maui "known" state:

https://github.com/dotnet/maui/blob/28e9cbb6d8e932fa04302d2e42bb8003dfe80d4d/src/Controls/src/Core/VisualStateManager.cs#L15-L23

This is also documented: https://learn.microsoft.com/dotnet/maui/user-interface/visual-states

The main cause of the switch is that when controls are [un]focused/[en|dis|abled, they switch states. In the case of this example, I found they would switch from `Focused` to `Notrmal`.

If you use another word for the "nortmal" state - like `Wide`, then everything works as expected.
